# For chatbot interview prep we are decide to work on 6 job roles.
* 1- Sales-manager
* 2-Human resources
* 3-Customer-Support
* 4-Business-development
* 5-Account-Manager
* 6-Account-Executive

# Collected data from data collecting team

# Import library for reading data

In [ ]:
import numpy as np
import pandas as pd

## Read and view sales manager dataset

In [ ]:
sales_manager=pd.read_json("sales_manager_interview_questions.json")

In [ ]:
sales_manager.head()

date company_name                                          job_title  \
0 2015-10-27      Aaron's               Assistant Sales Manager was asked...   
1 2020-04-15  WhiteHat Jr                         Sales Manager was asked...   
2 2012-08-01     Etisalat  Sales Customer Service Representative was aske...   
3 2013-12-27           NA                 Channel Sales Manager was asked...   
4 2016-09-26    Smartfren                 Sales Program Manager was asked...   

                    interview_question  
0     What are your management skills?  
1                      Why whitehatjr?  
2         i am fine and have no issues  
3            Can u Join Immediately???  
4  Could you ready placed in branches?

In [ ]:
# data info()
sales_manager.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3089 entries, 0 to 3088
Data columns (total 4 columns):
 #   Column              Non-Null Count  Dtype         
---  ------              --------------  -----         
 0   date                3089 non-null   datetime64[ns]
 1   company_name        3089 non-null   object        
 2   job_title           3089 non-null   object        
 3   interview_question  3089 non-null   object        
dtypes: datetime64[ns](1), object(3)
memory usage: 96.7+ KB


# Data description of sales_manager
* Shape of data 3089 rows and 4 columns.
* Null values is not present in this dataset
* Columns description
    * Date- Date column have date information on which date question is aksed.
    * Company_name- Company name column have company name information.
    * Job_title- job title column have profile information.
    * Interview questions- This column have interview questions which is asked in interview.

# Read and view HR dataset

In [ ]:
hr=pd.read_json("hr_interview_questions.json")

In [ ]:
hr.head()

date                                  company_name  \
0 2015-03-29                                      Alokozay   
1 2012-09-28  Mercedes-Benz Research and Development India   
2 2022-08-31                                   Weatherford   
3 2017-07-13                                       Bedrock   
4 2019-03-10                    Visions Personnel Services   

                                        job_title  \
0                    Human Resources was asked...   
1           Director Human Resources was asked...   
2                         HR Advisor was asked...   
3  Vice President of Human Resources was asked...   
4      Human Resources Administrator was asked...   

                                  interview_question  
0             Q: Are you open to working long hours?  
1  How will ensure that attrition is with accepta...  
2                      Tell me about your experience  
3  Asked what I would have done if the CEO asked ...  
4                            tell me about your self

In [ ]:
hr.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 3428 entries, 0 to 3427
Data columns (total 4 columns):
 #   Column              Non-Null Count  Dtype         
---  ------              --------------  -----         
 0   date                3428 non-null   datetime64[ns]
 1   company_name        3428 non-null   object        
 2   job_title           3428 non-null   object        
 3   interview_question  3428 non-null   object        
dtypes: datetime64[ns](1), object(3)
memory usage: 107.2+ KB


# Data description of hr dataset
* Shape of data is 3428 rows and 4 columns.
* Null values is not present in this dataset.
* Columns have same information like sales-manager dataset.

# Read and view customer_support dataset

In [ ]:
customer_support=pd.read_json("customer_support_interview_questions.json")

In [ ]:
customer_support.head()

date company_name                                          job_title  \
0 2015-12-26   Thermotron        Customer Support Field Service was asked...   
1 2013-11-27      OSIsoft             Customer Support Engineer was asked...   
2 2012-01-07     LinkedIn  Global Customer Support Representative was ask...   
3 2012-10-20      Mphasis                      Customer Support was asked...   
4 2015-04-01       Intuit  Customer Service and Support Specialist was as...   

                                  interview_question  
0         have you ever been fired or asked to leave  
1  Question 1: You have a railroad track that run...  
2  You’re in a room with three light switches, ea...  
3  most of difficult is that whom guys working in...  
4             How do you screen shot using the iMac?

In [ ]:
customer_support.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 29083 entries, 0 to 29082
Data columns (total 4 columns):
 #   Column              Non-Null Count  Dtype         
---  ------              --------------  -----         
 0   date                29083 non-null  datetime64[ns]
 1   company_name        29083 non-null  object        
 2   job_title           29083 non-null  object        
 3   interview_question  29083 non-null  object        
dtypes: datetime64[ns](1), object(3)
memory usage: 909.0+ KB


# data description of customer dataset
* Shape of the data 29083 rows and 4 columns.
* Null values is not presents in this dataset.
* Columns have same information like sales manager dataset.

# Read and view business development dataset


In [ ]:
business_development=pd.read_json("customer_support_interview_questions.json")

In [ ]:
business_development.head()

date company_name                                          job_title  \
0 2015-12-26   Thermotron        Customer Support Field Service was asked...   
1 2013-11-27      OSIsoft             Customer Support Engineer was asked...   
2 2012-01-07     LinkedIn  Global Customer Support Representative was ask...   
3 2012-10-20      Mphasis                      Customer Support was asked...   
4 2015-04-01       Intuit  Customer Service and Support Specialist was as...   

                                  interview_question  
0         have you ever been fired or asked to leave  
1  Question 1: You have a railroad track that run...  
2  You’re in a room with three light switches, ea...  
3  most of difficult is that whom guys working in...  
4             How do you screen shot using the iMac?

In [ ]:
business_development.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 29083 entries, 0 to 29082
Data columns (total 4 columns):
 #   Column              Non-Null Count  Dtype         
---  ------              --------------  -----         
 0   date                29083 non-null  datetime64[ns]
 1   company_name        29083 non-null  object        
 2   job_title           29083 non-null  object        
 3   interview_question  29083 non-null  object        
dtypes: datetime64[ns](1), object(3)
memory usage: 909.0+ KB


# Data description of business developments
* Shape of the data 29083 rows and 4 columns.
* Null values is not present in this dataset.
* Columns have same information like sales-manager dataset.

# Read and view account manager dataset.

In [ ]:
account_manager=pd.read_json("account_manager_interview_questions.json")

In [ ]:
account_manager.head()

date                  company_name  \
0 2015-02-08                         H E B   
1 2013-08-21                           IRS   
2 2009-09-10  Affiliated Computer Services   
3 2009-05-05                           NCR   
4 2014-10-21                          Yelp   

                                      job_title  \
0  Customer Service Representative was asked...   
1  Customer Service Representative was asked...   
2  Customer Service Representative was asked...   
3         National Account Manager was asked...   
4         Account Manager/New Grad was asked...   

                                  interview_question  
0  How would you respond to a customer that is co...  
1  How would you deal with screaming customers or...  
2  What do you consider exceptional customer serv...  
3  Describe a sales situation where price seemed ...  
4  If I was David's pizza and I wanted to cancel ...

In [ ]:
account_manager.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9953 entries, 0 to 9952
Data columns (total 4 columns):
 #   Column              Non-Null Count  Dtype         
---  ------              --------------  -----         
 0   date                9953 non-null   datetime64[ns]
 1   company_name        9953 non-null   object        
 2   job_title           9953 non-null   object        
 3   interview_question  9953 non-null   object        
dtypes: datetime64[ns](1), object(3)
memory usage: 311.2+ KB


# Data description of account_manager
* Shape of the data 9953 rows and 4 columns.
* Null values is not present in this dataset.
* Columns have same information like sales manager dataset.

# Read and view account_executive dataset


In [ ]:
account_executive=pd.read_json("account_executive_interview_questions .json")

In [ ]:
account_executive.head()

date  company_name                                          job_title  \
0 2012-11-19          Epic                     Account Executive was asked...   
1 2014-10-03     Glassdoor          Enterprise Account Executive was asked...   
2 2011-05-02          Meta                     Account Executive was asked...   
3 2012-07-06  KMV Projects                    Accounts Executive was asked...   
4 2012-06-30        MAERSK  Senior Executive - Finance & Accounts was aske...   

                                  interview_question  
0  A potential, very potentially profitable, clie...  
1  What is the most important part of the sales c...  
2  If you were handed 50 new target accounts how ...  
3                                 Do you know Hindi?  
4  What would you do, if you are in a situation w...

In [ ]:
account_executive.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 16101 entries, 0 to 16100
Data columns (total 4 columns):
 #   Column              Non-Null Count  Dtype         
---  ------              --------------  -----         
 0   date                16101 non-null  datetime64[ns]
 1   company_name        16101 non-null  object        
 2   job_title           16101 non-null  object        
 3   interview_question  16101 non-null  object        
dtypes: datetime64[ns](1), object(3)
memory usage: 503.3+ KB


# Data description of account executive dataset
* Shape of the data 16101 rows and 4 columns.
* Null values is not present in this dataset.
* Columns have same information like sales manager.

# Combining all the files into one dataframe for the ease of processing with help of concat() function

In [ ]:
df=pd.concat([sales_manager,hr,business_development,account_executive,account_manager,customer_support],ignore_index=True)

# Creating a backup of this dataframe

In [ ]:
backup=df.copy()

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 90737 entries, 0 to 90736
Data columns (total 4 columns):
 #   Column              Non-Null Count  Dtype         
---  ------              --------------  -----         
 0   date                90737 non-null  datetime64[ns]
 1   company_name        90737 non-null  object        
 2   job_title           90737 non-null  object        
 3   interview_question  90737 non-null  object        
dtypes: datetime64[ns](1), object(3)
memory usage: 2.8+ MB


df have 90737 rows and 4 columns , null values is not present in this dataset.

# Importing library for removing non english words

In [ ]:
!pip install langdetect

     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 981.5/981.5 kB 13.3 MB/s eta 0:00:00
  Preparing metadata (setup.py) ... done
  Created wheel for langdetect: filename=langdetect-1.0.9-py3-none-any.whl size=993225 sha256=f9ee5a92c0b0cce88eeed650b85019364fe9cc779a0917aaeacfd443468c6ed2
  Stored in directory: /root/.cache/pip/wheels/95/03/7d/59ea870c70ce4e5a370638b5462a7711ab78fba2f655d05106
Successfully built langdetect


In [ ]:
from langdetect import detect


# Function for removing non-english questions

In [ ]:
def detect_english(interview_question):
  try:
    return detect(interview_question)=='en'
  except:
    return False

# Apply function on dataset

In [ ]:
df=df[df['interview_question'].apply(detect_english)]

# Resetting indexes

In [ ]:
df.reset_index(inplace=True,drop=True)

In [ ]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 81623 entries, 0 to 81622
Data columns (total 4 columns):
 #   Column              Non-Null Count  Dtype         
---  ------              --------------  -----         
 0   date                81623 non-null  datetime64[ns]
 1   company_name        81623 non-null  object        
 2   job_title           81623 non-null  object        
 3   interview_question  81623 non-null  object        
dtypes: datetime64[ns](1), object(3)
memory usage: 2.5+ MB


Total [90737-81655]=9082 rows removed.New shape of data 81655 rows and 4 columns.

# Remove non-readable characters.

In [ ]:
df['interview_question']=df['interview_question'].str.replace('\W',' ',regex=True)

In [ ]:
df.head()

date company_name                                          job_title  \
0 2015-10-27      Aaron's               Assistant Sales Manager was asked...   
1 2020-04-15  WhiteHat Jr                         Sales Manager was asked...   
2 2012-08-01     Etisalat  Sales Customer Service Representative was aske...   
3 2013-12-27           NA                 Channel Sales Manager was asked...   
4 2016-09-26    Smartfren                 Sales Program Manager was asked...   

                    interview_question  
0     What are your management skills   
1                      Why whitehatjr   
2         i am fine and have no issues  
3            Can u Join Immediately     
4  Could you ready placed in branches

# Remove was asked... from job title column

In [ ]:
df['job_title']=df['job_title'].str.replace('was asked...',' ')

<ipython-input-32-c208028992b7>:1: FutureWarning: The default value of regex will change from True to False in a future version.
  df['job_title']=df['job_title'].str.replace('was asked...',' ')


In [ ]:
df.head()

date company_name                                job_title  \
0 2015-10-27      Aaron's                Assistant Sales Manager     
1 2020-04-15  WhiteHat Jr                          Sales Manager     
2 2012-08-01     Etisalat  Sales Customer Service Representative     
3 2013-12-27           NA                  Channel Sales Manager     
4 2016-09-26    Smartfren                  Sales Program Manager     

                    interview_question  
0     What are your management skills   
1                      Why whitehatjr   
2         i am fine and have no issues  
3            Can u Join Immediately     
4  Could you ready placed in branches

In [ ]:
df.to_csv("only_english_data.csv")